lib/puppet/type/zone.rb

```diff
@@ -135,6 +151,10 @@ def up?
 	  sysconfig SCI tool on first boot of the zone."
   end
 
+  newparam(:zonecfg_archive) do
+    desc "Configure a zone from an Unified Archive"
+  end
+
   newparam(:zonecfg_export) do
     desc "Contains the zone configuration information. This can be passed in
     in the form of a file generated by the zonecfg command, in the form
```

Comment:
validate this input to either be a file or a more reasonably valid set of commands, something like the following.
```
validate do |value|
unless value.match((%r(^/),/\n|;|(?:commit|exit)$/)
  fail "#{value} does not look like an absolute path or a set of zoncfg commands"
end
```
or
Set parameter input values.
`newvalues(%r(^/),/\n|;|(?:commit|exit)$/'

Validate is probably the better option as it gives you more control over the error message.

Also update the desc to indicate that the zonecfg_export file must be local to the managed system. 

For extra credit add an autorequire on File resources for the value of zonecfg_export. 
[See: autorequire(:file)](https://github.com/shawnferry/puppet-solaris_providers/blob/sferry/ilb-vrrp-zfsacl/lib/puppet/type/zfs_acl.rb#L291) here for an example